draft-jmiller-json-web-proof.md

```diff
@@ -121,44 +98,27 @@ For example:
 }
 ```
 
+Every JWP algorithm must include a digest method that is used to generate a hash of the base64url serialized protected header.  The protected header cannot be selectively disclosed and the digest value MUST be included in all proof values.
+
 ## Payloads
 
-Payloads are always represented as an ordered array.  Each payload's content type is either known by context or specified in that payload's associated header.  The default content type is `application/json`, with the JSON object values being JWT claims.
+Payloads are always represented as an ordered array.  The mapping of which value is in which payload slot is out of scope of this specification.
 
-Example payloads:
-```json
-{
-    "given_name":"Jane",
-    "family_name":"Doe"
-}
-```
-```json
-{
-    "email":"janedoe@example.com"
-}
-```
-```json
-{
-    "birthdate":"0000-03-22"
-}
-```
+In order to support ZKPs, individual payloads cannot be serialized before they are passed into an algorithm implementation.  This enables the algorithms to accept and internally encode elliptic curve points, blinded values, plain numbers, membership keys, etc.  Implementations are therefore required to provide optional arguments for each payload such that the application can utilize these capabilities as needed.
 
+When selective disclosure preferences are applied, any one or more payloads may be hidden.  The position of other payloads does not change due to any preceeding ones being hidden - the resulting array will simply be sparse without the hidden payloads.
```

Comment:
This sentence does not make sense to me... are you meaning to say that the length of `payload: Array<ANY>` is constant / final and hidding values is like doing this:

items[disclosureIndex'] = 'hidden' ?

Reply:
Pushed some improvements here that hopefully clarify also.